Sources/Host/Linux/PTrace.cpp

```diff
@@ -255,19 +255,16 @@ ErrorCode PTrace::suspend(ProcessThreadId const &ptid) {
 
 ErrorCode PTrace::step(ProcessThreadId const &ptid, ProcessInfo const &pinfo,
                        int signal, Address const &address) {
+#if defined(ARCH_ARM)
+  DS2BUG("single stepping for ARM must be implemented in software");
+#endif
```

Comment:
I thought about this, but then we have to alter the headers as well, and
I'd love to avoid having more `#ifdefs` than we already have. Not sure if
there's a cleaner way to do it at this point. I'll take a look though,
compile error would be better here.

On Mon, Apr 18, 2016 at 12:24 PM, strager <notifications@github.com> wrote:

> In Sources/Host/Linux/PTrace.cpp
> <https://github.com/facebook/ds2/pull/282#discussion_r60118811>:
>
> > @@ -255,19 +255,16 @@ ErrorCode PTrace::suspend(ProcessThreadId const &ptid) {
> >
> >  ErrorCode PTrace::step(ProcessThreadId const &ptid, ProcessInfo const &pinfo,
> >                         int signal, Address const &address) {
> > +#if defined(ARCH_ARM)
> > +  DS2BUG("single stepping for ARM must be implemented in software");
> > +#endif
>
> What if we make this a compilation error by commenting out the entire
> definition on ARM?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/282/files/8bcd2b3177be09d93ae1335b054bb7a354d0f5b4..55c8a56c6d24acd45b663fe1f5b0044bf5f10a6e#r60118811>
>



-- 
Stephane Sezer